torch/_inductor/codecache.py

```diff
@@ -1157,6 +1159,207 @@ def load(cls, kernel_name, source_code):
+        options.extend(["-lineinfo", "-g -G", "-DCUTLASS_DEBUG_TRACE_LEVEL=1"])
+    if config.cuda.enable_ptxas_info:
+        options.extend(
+            [
+                "--keep",  # Keep the intermediate files for debugging (including ptx, sass, cubin etc.)
+                "--ptxas-options=--warn-on-local-memory-usage",  # warn us if local memory is used in CUDA Kernels
+                "--ptxas-options=--warn-on-spills",  # warn us if register spilling happens in CUDA Kernels
+                "--resource-usage",  # Report on CUDA resource usage (shared mem, registers etc.)
+                "--source-in-ptx",
+            ]
+        ),  # Annotate the ptx file with source information
+    if config.cuda.use_fast_math:
+        options.extend(
+            [
+                "--use_fast_math",
+                "-DCUTLASS_USE_TANH_FOR_SIGMOID=1",
+            ]
+        )
+    return options
+
+
+def cuda_compile_command(
+    src_files: List[str],
+    dst_file: str,
+    dst_file_suffix: str,
+) -> str:
+    include_paths = _cutlass_include_paths()
+    nvcc_host_compiler_options = _nvcc_host_compiler_options()
+    nvcc_compiler_options = _nvcc_compiler_options()
+    options = (
+        nvcc_compiler_options
+        + [
+            f"-Xcompiler {opt}" if "=" in opt else f"-Xcompiler={opt}"
+            for opt in nvcc_host_compiler_options
+        ]
+        + ["-I" + path for path in include_paths]
+    )
+    src_file = " ".join(src_files)
+    res = ""
+    if dst_file_suffix == "o":
+        res = (
+            _cuda_compiler() + " " + " ".join(options) + f" -c -o {dst_file} {src_file}"
+        )
+    elif dst_file_suffix == "so":
+        options.append("-shared")
+        res = _cuda_compiler() + " " + " ".join(options) + f" -o {dst_file} {src_file}"
+    else:
+        raise NotImplementedError(f"Unsupported output file suffix {dst_file_suffix}!")
+    print(f"CUDA command: {res}")
+    return res
+
+
+class DLLWrapper:
+    def __init__(
+        self,
+        lib_path: str,
+    ):
+        self.lib_path = lib_path
+        self.DLL = ctypes.cdll.LoadLibrary(lib_path)
+        self.is_open = True
+
+    def close(self):
+        if self.is_open:
+            self._dlclose()
+            self.is_open = False
+
+    def _dlclose(self):
+        f_dlclose = None
+
+        if is_windows():
+            f_dlclose = ctypes.windll.kernel32.FreeLibrary
+        elif is_linux():
+            syms = ctypes.CDLL(None)
+            if not hasattr(syms, "dlclose"):
+                # Apline Linux
+                syms = ctypes.CDLL("libc.so")
+
+            if hasattr(syms, "dlclose"):
+                f_dlclose = syms.dlclose
+
+        if f_dlclose is not None:
+            f_dlclose.argtypes = [ctypes.c_void_p]
+            f_dlclose(self.DLL._handle)
+        else:
+            logging.warning(
+                "dll unloading function was not found, library may not be unloaded properly!"
+            )
```

Comment:
Yeah I copied them from cpp_extension and AITemplate code. After a second thought I think I'll just remove "windows" logics since they're not tested. People could work on more proper window support later.